Fix/Workaround + Feature

bactaholic opened the pull request:
Provides workaround for #122 allowing you to toggle the swipe gestures in the general settings page, also adds double tap on nav bar feature to replace it for people like me who still want quick access. In my testing both work seamlessly separate or together.

Additionally, I'm not sure if it was removed on purpose but I really liked the original total post/comments counter from the original alpha. I added that back in to the profile header and added a toggle in the settings to swap post/comments counts to post/comments score. I can remove this if it was removed intentionally.

hjiangsu commented:
Thanks! I took a quick look over it and it seems to look good - just a question:
- How does the double-tap interact with the options in the bottom nav bar? would you have to double tap outside of the icons in the bottom nav bar to trigger the drawer to open/close?

> Additionally, I'm not sure if it was removed on purpose but I really liked the original total post/comments counter from the original alpha. I added that back in to the profile header and added a toggle in the settings to swap post/comments counts to post/comments score. I can remove this if it was removed intentionally.

I removed it because I was working on introducing a better user profile page! This is all happening in the `develop` branch, and will be merged into `main` when I release the upcoming version (which would be sometime tonight for me)

What I would suggest is to rebase this to the develop branch so that its up to date with all the changes with user profiles that I've added. The second reason why I removed it was because it didn't really seem too clear to me at least what the numbers were representing. I was trying to figure out a better way to represent those numbers (e.g., which one is post count and which one is comment count)

Aside from that, great work and thank you for the contributions! I'll most likely merge this in for the next-next release if thats okay (that would be v0.2.1+10)

bactaholic commented:
Howdy!

Thanks for the feedback, I will rebase this first thing in the morning as it's late and my laptop is dead.

As far as the double tap functionality, when turned on with the home page as the active window you can tap anywhere along the taskbar, including on the icons, and it will open the sidebar. I havent had any issues with it accidentally selecting a different page instead of activing the side bar, the tap delay seems to be long enough.
It also doesn't interfere with the existing swipe command. In fact, if you have both enabled, you can swipe to open and double tap to close, or vice versa. They appear to work interchangeably with no issues but I've only tested on one device, on Android.

I'll remove my profile score/counters from the rebase since you're working on the user profile. I've also thought about how to differentiate the different scores. I thought about simply adding 
"P: [postScore] C: [commentScore] T: [total]"
But I wasn't totally sure if that was clear enough. Will give it more thought. Hopefully I'll have some more fixes and ideas before the next-next release.

Thanks!

hjiangsu commented:
> Thanks for the feedback, I will rebase this first thing in the morning as it's late and my laptop is dead.

Actually - I think your changes might just be fine with the `develop` branch, so don't worry about rebasing it! I just merged in develop into main to prep the the upcoming release, and there's no conflicts that I see :D

> As far as the double tap functionality, when turned on with the home page as the active window you can tap anywhere along the taskbar, including on the icons, and it will open the sidebar. 

Ahh okay okay, that seems to make sense! Thanks!

> I'll remove my profile score/counters from the rebase since you're working on the user profile. 

I think for now, it might be best to remove it, but I do want to re-introduce the idea of showing the scores in a future version since I think it's useful to have that information. But yeah, as you mentioned, I do want to make it a bit more obvious what the numbers represent (either through icons, or some sort of text/visual indicator)

bactaholic commented:
@hjiangsu howdy! Removed the counters and associated switch in settings, also noticed I missed a line and it was causing the swipe gestures button to not retain its position after you switch pages, fixed that. Good to go.

hjiangsu commented:
Thanks! I'll check it out later when I have the time 😁

hjiangsu commented:
Looks good! If I were to nitpick, I would just suggest using more descriptive names for the settings. 

`swipeGestures` and `doubleTapGestures` is a bit difficult to know what exactly it affects! For example, maybe `bottomNavBarSwipeGestures` and `bottomNavBarDoubleTapGestures` would be more appropriate!

bactaholic commented:
Sure thing, easy fix. Good to go.

hjiangsu commented:
It looks good! Thanks :D

hjiangsu commented:
Just a quick update on something - I've temporarily disabled the double tap to open sidebar. There was a issue where I noticed a longer delay between navigating the bottom nav bar when the double-tap gesture was implemented.

This is most likely because `GestureDetector` increased the delay of a single tap to be longer in order to detect cases of double-tapping. I'll take a look how we can maybe still have the same responsiveness feel while also including the double-tap functionality!

bactaholic commented:
> Just a quick update on something - I've temporarily disabled the double tap to open sidebar. There was a issue where I noticed a longer delay between navigating the bottom nav bar when the double-tap gesture was implemented.
> 
> This is most likely because `GestureDetector` increased the delay of a single tap to be longer in order to detect cases of double-tapping. I'll take a look how we can maybe still have the same responsiveness feel while also including the double-tap functionality!

Yeah I noticed that slight delay as well, I got used to it pretty fast but I can see how it would be annoying. I've been thinking about ways to get around it, maybe there is a custom delay timer we can set. I'll be looking into this more as well.

hjiangsu commented:
Yeah - I was thinking maybe a compromise to this issue is to completely remove the double-tap gesture when the option is off, as opposed to keeping the double-tap there but checking if the setting is set in the function.

This would simply just swap around the logic: If the double-tap setting is false, then set the GestureDetector double-tap function to null, otherwise have the function as it at the moment

I think this might fix the issue for those who have it disabled, but those who have the double-tap enabled will have to readjust themselves to have a slightly longer delay

hjiangsu commented:
Just an update - I have made the changes I mentioned previously, and they seem to be working as intended!
- When double tap is disabled, the delay between tapping an icon and the action is what it was previously
- When double tap is enabled, there is a slightly longer delay between tapping to allow the double tap gesture to register
